Classes/Profile/ProfileLoadingComponent.swift

```diff
@@ -0,0 +1,56 @@
+//
+//  ProfileComponent.swift
+//  Freetime
+//
+//  Created by Sash Zats on 2/3/18.
+//  Copyright © 2018 Ryan Nystrom. All rights reserved.
+//
+
+import Theodolite
+
+enum State {
+  case idle
+  case loading
+  case failed(error: Error)
+  case succeeded(ProfileResult)
+}
+
+final class ProfileLoadingComponent: Component, TypedComponent {
+  typealias PropType = (GithubClient)
+
+  typealias StateType = State
```

Comment:
Should we bump the State type inside the component?

Reply:
Not sure I understand your point here